chain/actors/builtin/miner/actor.go.template

```diff
@@ -334,35 +336,75 @@ func AllCodes() []cid.Cid {
-		networkQaPowerEstimate builtin.FilterEstimate,
-		qaSectorPower abi.StoragePower,
+        nwVer network.Version,
+        rewardEstimate builtin.FilterEstimate,
+        networkQaPowerEstimate builtin.FilterEstimate,
+        qaSectorPower abi.StoragePower,
 ) (abi.TokenAmount, error) {
-	v, err := actorstypes.VersionForNetwork(nwVer)
-	if err != nil {
-		return big.Zero(), err
-	}
-	switch v {
-	{{- range .versions}}
-	case actorstypes.Version{{.}}:
-		{{- if (ge . 16)}}
-			return minertypes{{.}}.PledgePenaltyForContinuedFault(
-				smoothing{{.}}.FilterEstimate{
-					PositionEstimate: rewardEstimate.PositionEstimate,
-					VelocityEstimate: rewardEstimate.VelocityEstimate,
-				},
-				smoothing{{.}}.FilterEstimate{
-					PositionEstimate: networkQaPowerEstimate.PositionEstimate,
-					VelocityEstimate: networkQaPowerEstimate.VelocityEstimate,
-				},
-				qaSectorPower,
-			), nil
-		{{- else}}
-				return big.Zero(), nil
-		{{- end -}}
-	{{end}}
-	default:
-		return big.Zero(), xerrors.Errorf("unsupported network version")
-	}
+    v, err := actorstypes.VersionForNetwork(nwVer)
+    if err != nil {
+        return big.Zero(), err
+    }
+
+    if v <= actorstypes.Version16 {
+        return minertypes16.PledgePenaltyForContinuedFault(
+            smoothing16.FilterEstimate{
+                PositionEstimate: rewardEstimate.PositionEstimate,
+                VelocityEstimate: rewardEstimate.VelocityEstimate,
+            },
+            smoothing16.FilterEstimate{
+                PositionEstimate: networkQaPowerEstimate.PositionEstimate,
+                VelocityEstimate: networkQaPowerEstimate.VelocityEstimate,
+            },
+            qaSectorPower,
+        ), nil
+    }
+
+    switch v {
+    {{- range .versions}}
+        {{- if (gt . 16)}}
+    case actorstypes.Version{{.}}:
+            return minertypes{{.}}.PledgePenaltyForContinuedFault(
+                smoothing{{.}}.FilterEstimate{
+                    PositionEstimate: rewardEstimate.PositionEstimate,
+                    VelocityEstimate: rewardEstimate.VelocityEstimate,
+                },
+                smoothing{{.}}.FilterEstimate{
+                    PositionEstimate: networkQaPowerEstimate.PositionEstimate,
+                    VelocityEstimate: networkQaPowerEstimate.VelocityEstimate,
+                },
+                qaSectorPower,
+            ), nil
+        {{- end}}
+    {{- end}}
+    default:
+        return big.Zero(), xerrors.Errorf("unsupported network version: %d", v)
+    }
 }
+
+func PledgePenaltyForTermination(
+        nwVer network.Version,
+        initialPledge abi.TokenAmount,
+        sectorAge abi.ChainEpoch,
+        faultFee abi.TokenAmount,
+) (abi.TokenAmount, error) {
+    v, err := actorstypes.VersionForNetwork(nwVer)
+    if err != nil {
+        return big.Zero(), err
+    }
+
+    if v < actorstypes.Version16 {
+        return big.Zero(), xerrors.Errorf("not supported for network version %d", v)
+    }
```

Comment:
Is this necessary? We already have a default below.

Reply:
I guess not, I was imagining the different language would be more helpful than it actually is

removed